### ISSUE TITLE
Make oemof more user friendly

### ISSUE THREAD

**Issue by birgits** [oemof/organisation#17]
Since the user meeting there has been some discussion about ways to make the first steps in oemof easier and how to connect users. I tried to collect the different ideas here. We can discuss them at the next webco in June. 

* oemof book
* separate user and developer documentation
* take the existing examples (https://github.com/oemof/oemof/tree/dev/examples/solph) add some documentation and move them to the main documentation (as PyPSA does it https://www.pypsa.org/examples/index.html) or a separate documentation; examples could also be the basis for a paper that could be linked on our website
* jupyter notebooks for examples
* improved documentation on how to run the examples for python newbies
* user page on homepage with link to projects using oemof, user forum, youtube videos
* FAQ on homepage
* user forum at openmod forum

**Comment by busiing** [comment]
"Push" from developer meeting 2018:

- jupyter tutorials (Uwe?)
- video tutorials (Jakob?)

**Comment by uvchik** [comment]
Sorry, it was probably a misunderstanding. I do not know why my name was written there. I crossed it out.

**Comment by busiing** [comment]
Just had a quick Call from @jakob-wo. We'll try to start some brainstorming on necessary tutorial videos. Since his research project ends soon it might be a (wo)men-power problem to create the videos.

**Comment by jakob-wo** [comment]
Hi everyone,
unfortunately I won't be able to distribute any video-tutorials. As @busiing mentioned, our research project ends this month.
However I still think video-tutorial might be a crucial aspect to improve user-friendliness for beginners. I hope someone will carry on with this idea!

**Comment by Bachibouzouk** [comment]
Hello everyone,
As I could see that many stakeholders run away or are scared when they see a single line of code, I think a way to make oemof more accessible to a larger number of people is to allow non-programmers to build energy system models via an interface (possibility to drag and drop components to build the energy system model for example).

Fortunately, in the context of the open_plan project at RLI (https://reiner-lemoine-institut.de/open_plan_2020/, https://github.com/rl-institut/open_plan), one of our goal is to develop a tool for non-programmers to build energy system models (the tool will be based on oemof in the backend). 
I think that it could be a good opportunity for oemof dev and oemof users to speak out and exchange at the beginning of the development of the tool about potential features which could be implemented which they could re-use then. 

In short, we have to develop the tool anyway, but we might as well try to make certain parts of it modular enough so that other people can use them for their own projects. And maybe create a repo within oemof if it makes sense?

**Comment by fwitte** [comment]
Hi @Bachibouzouk,
I think this would be a great addition. This would definitively be a nice feature for tespy, too. Will you participate in the dev meeting this week?
Best
Francesco

**Comment by Bachibouzouk** [comment]
> Will you participate in the dev meeting this week #80?

Depending when my child decides to come to the world: yes or now :)

**Comment by Bachibouzouk** [comment]
> > Will you participate in the dev meeting this week #80?
> 
> Depending when my child decides to come to the world: yes or now :) @c-moeller is also involved in open_plan and participates to the dev meeting for sure!

**Comment by p-snft** [comment]
As far as I know, it was planned to talk about this in the framework of the presentation on https://github.com/rl-institut/multi-vector-simulator/.

**Comment by Bachibouzouk** [comment]
@p-snft - open_plan will likely iterate ontop of https://github.com/rl-institut/multi-vector-simulator/. The latter does not necessarily have resources to include oemof users suggestions as the project is ending next year, whereas open_plan still has 2 years to go and could have resources for that (hence the proposition to discuss potential collaboration :))

**Comment by p-snft** [comment]
Maybe it's an option to propose an additional session on user experience when the meeting starts tomorrow. We have a backup slot on Thursday.

**Comment by fwitte** [comment]
> Depending when my child decides to come to the world
Congratulations! 
> Maybe it's an option to propose an additional session on user experience when the meeting starts tomorrow. We have a backup slot on Thursday.
Might be a good idea, let's see :)

**Comment by Bachibouzouk** [comment]
I think we could use the openmod forum questions about oemof as insipration to improve the documentation by providing usecases on top of component descriptions. 

As an example of what I mean: https://forum.openmod.org/t/investment-variation-investment-during-year/2677

There, the question is "Is there any way to vary the investment costs over time?" the the usecase is "Simulating a market until 2050"

As a user I would have these questions and if I search into the doc I might miss the answer if it is provided in a sub sub sub section

A minimal first step towards improvement would be to link the forum post from the docs (maybe under "User's guide" >>> "What questions/usecases I can address with oemof.solph") and a link to the examples relevant.

**Comment by Bachibouzouk** [comment]
The oemof-examples could be improved as well by providing a context or potential usecases, right now the descriptions at the top of the example modules focuse more on describing what component of oemof is in the example but not really what is the underlying usecase. Having the usecase in mind would help understand the example. A description of the results might be helpful too (like a figure caption in an article explaining what is to be seen in the figure and why it is interesting).

By usecase I mean the reallife problem one tries to solve using oemof.solph. To illustrate what I mean: the oemof-example `startup_shutdown` provides a usecase description `"""Example that illustrates how to model startup and shutdown costs attributed to a binary flow."""` 
To me it is not obvious what is the reallife usecase, the comments along the code below helped me further to understand what the example is about. I think it would make newcomers life easier if the description was linked with an explicit reallife example :)

**Comment by p-snft** [comment]
At the moment, we require a working example to add new functionality. (This rule applies e.g. to solph, not to all oemof projects). For this reason, the examples are tailored very specific to the (new) feature. In the last user meeting, I suggested to move the very code related examples into the corresponding repository (e.g. https://github.com/oemof/oemof-solph/).

My idea was to turn the examples project into something that would be more human readable. E.g. it could be compiled into a "learning by example" oemof book with an introduction, explanatory words what is be modelled in which example, and so on. Would that meet your suggestion?

**Comment by fwitte** [comment]
> My idea was to turn the examples project into something that would be more human readable. E.g. it could be compiled into a "learning by example" oemof book with an introduction, explanatory words what is be modelled in which example, and so on. Would that meet your suggestion?

I would totally agree with this idea! Providing supplementary material for the examples (fundamentals of the respective subject, maybe embedded in full 90 min lecture notes etc.) would be very beneficial as well (related to #94).
